src/main/java/com/waverleysoftware/listemers/TestListener.java

```diff
@@ -1,31 +1,41 @@
 package com.waverleysoftware.listemers;
 
 import lombok.extern.slf4j.Slf4j;
+import org.assertj.core.api.SoftAssertions;
 import org.testng.IInvokedMethod;
 import org.testng.IInvokedMethodListener;
 import org.testng.ITestResult;
+import org.testng.internal.TestResult;
 
-@SuppressWarnings("JavadocType")
+@SuppressWarnings({"JavadocType", "unchecked"})
 @Slf4j
 public class TestListener implements IInvokedMethodListener {
+    private static ThreadLocal thread = new ThreadLocal();
```

Comment:
make it `final`, and specify an exact type, you'll be working with as generic argument